spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -218,32 +250,44 @@ private SpannerStatementQueryExecutor() {
 				keyParts.add(parentKeyParts.next());
 				tagNum++;
 			}
-			orJoiner.add(andJoiner.toString());
+			orParts.add(andJoiner.toString());
 		}
-		String cond = orJoiner.toString();
+		String keyClause = orParts.size() == 1
```

Comment:
I do not agree with your argument. The code is already complex enough and this small update will not simplify it - it will just brake some IT tests. But, sure, I'll follow your advice.